nsxt/resource_nsxt_upgrade_run.go

```diff
@@ -756,10 +784,9 @@ func runUpgrade(upgradeClientSet *upgradeClientSet, partialUpgradeMap map[string
 	return nil
 }
 
-func runPostcheck(upgradeClient nsx.UpgradeClient, d *schema.ResourceData) {
-	for i := range upgradeComponentList {
-		component := upgradeComponentList[i]
-		if component != mpUpgradeGroup {
+func runPostcheck(upgradeClient nsx.UpgradeClient, d *schema.ResourceData, targetVersion string) {
+	for _, component := range getUpgradeComponentList(targetVersion) {
```

Comment:
Maybe lets make another list for postchecks, this way we will not need to pass new parameter to `runPostcheck`? Postchecks do not care about the `finalize` step.